pyproject.toml

```diff
@@ -51,15 +53,33 @@ select = [
     "PYI058", # use `Iterator` as the return type for `__iter__` methods
 ]
 ignore = [
+    ###
+    # Rules we don't want or don't agree with
+    ###
     # Slower and more verbose https://github.com/astral-sh/ruff/issues/7871
     "UP038", # Use `X | Y` in `isinstance` call instead of `(X, Y)`
+    # Used for direct, non-subclass type comparison, for example: `type(val) is str`
+    # see https://github.com/astral-sh/ruff/issues/6465
+    "E721", # Do not compare types, use `isinstance()`
+    ###
+    # False-positives, but already checked by type-checkers
+    ###
+    # Ruff doesn't support multi-file analysis yet: https://github.com/astral-sh/ruff/issues/5295
+    "RUF013", # PEP 484 prohibits implicit `Optional`
 ]
-
 [tool.ruff.lint.per-file-ignores]
 "*.pyi" = [
     # Most flake8-bugbear rules don't apply for third-party stubs like typeshed.
     # B033 could be slightly useful but Ruff doesn't have per-file select
     "B", # flake8-bugbear
+    # Rules that are out of the control of stub authors:
+    "E501", # Line too long
+    "E741", # ambiguous variable name
+    "F403", # `from . import *` used; unable to detect undefined names
+    # False positives in stubs
+    "F821", # Undefined name: https://github.com/astral-sh/ruff/issues/3011
+    # FIXME: Fix these in stubs, then remove these entries
+    "F405", # may be undefined, or defined from star imports
```

Comment:
These appear in `stdlib\_pydecimal.pyi` and `stdlib\ast.pyi`
Is that a pattern we'd like to change as my comment implies here? Or disabled specifically in those two files. Or is using a name from a star import something we wanna continue doing in typeshed (I feel like explicit would be better than implicit, even if we're talking entire module re-exports: do the `*` import + explicitly import symbols used in the stub)

Reply:
I don't think F405 is useful in stubs, type checkers will catch anything it catches.